File: AndroidApp/src/main/java/com/smp/sharedkotlin/app/data/MemoEntityMapper.kt
package com.smp.sharedkotlin.app.data

import com.smp.sharedkotlin.app.support.Mapper
import com.smp.sharedkotlin.entity.MemoEntity

object MemoEntityMapper : Mapper<MemoModel, MemoEntity> {

    override fun map(obj: MemoModel): MemoEntity {
        return MemoEntity(
            obj.id,
            obj.title,
            obj.content,
            obj.lastModifiedAt,
            obj.createdAt
        )
    }

    override fun mapAll(objs: Iterable<MemoModel>): List<MemoEntity> {
        return objs.map { map(it) }
    }

    override fun reverse(obj: MemoEntity): MemoModel {
        return MemoModel(
            obj.id,
            obj.title,
            obj.content,
            obj.lastModifiedAt,
            obj.createdAt
        )
    }

    override fun reverseAll(objs: Iterable<MemoEntity>): List<MemoModel> {
        return objs.map { reverse(it) }
    }

}

File: iOSApp/iOSApp/New Group/MemoEntityMapper.swift
import SharedCode

class MemoEntityMapper : Mapper {

    func map(obj: MemoModel) -> MemoEntity {
        return MemoEntity(id: obj.id, title: obj.title, content: obj.content, createdAt: obj.createdAt, lastModifiedAt: obj.lastModifiedAt)
    }
    
    func mapAll(objs: [MemoModel]) -> [MemoEntity] {
        return objs.map { map(obj: $0) }
    }
    
    func reverse(obj: MemoEntity) -> MemoModel {
        return MemoModel(id: obj.id, title: obj.title, content: obj.content, createdAt: obj.createdAt, lastModifiedAt: obj.lastModifiedAt)
    }
    
    func reverseAll(objs: [MemoEntity]) -> [MemoModel] {
        return objs.map { reverse(obj: $0) }
    }
}
